lib/pp.ml

```diff
@@ -297,3 +311,66 @@ let prvect_with_sep sep elem v = prvecti_with_sep sep (fun _ -> elem) v
 let prvect elem v = prvect_with_sep mt elem v
 
 let surround p = hov 1 (str"(" ++ p ++ str")")
+
+(*** DEBUG code ***)
+
+let db_print_pp ?(fmt=Format.std_formatter) pp =
+  let open Format in
+  let block_type btype =
+    let (bt, v) =
+      match btype with
+      | Pp_hbox v -> ("Pp_hbox", v)
+      | Pp_vbox v -> ("Pp_vbox", v)
+      | Pp_hvbox v -> ("Pp_hvbox", v)
+      | Pp_hovbox v -> ("Pp_hovbox", v)
+    in
+    Printf.sprintf "%s %d" bt v
```

Comment:
This should not return a string but a formatter function.

Reply:
The OCaml documentation is misleading.  `%a` is only described in the Printf doc, where it takes an `outchan`, but for Format.fprintf, it should take a `formatter`.

```
a: user-defined printer. Take two arguments and apply the first one to outchan (the current 
output channel) and to the second argument. The first argument must therefore have type 
out_channel -> 'b -> unit and the second 'b. The output produced by the function is inserted
in the output of fprintf at the current point.
```

Reply:
Yeah, I found the documentation on how to use `Format` terrible. Sorry I am not more helpful but I am quite in a rush these days.